x-pack/plugins/monitoring/public/application/pages/logstash/pipeline.tsx

```diff
@@ -0,0 +1,151 @@
+    },
+  });
+
+  const getPageData = useCallback(async () => {
+    const url = pipelineHash
+      ? `../api/monitoring/v1/clusters/${clusterUuid}/logstash/pipeline/${pipelineId}/${pipelineHash}`
+      : `../api/monitoring/v1/clusters/${clusterUuid}/logstash/pipeline/${pipelineId}`;
+
+    const response = await services.http?.fetch(url, {
+      method: 'POST',
+      body: JSON.stringify({
+        ccs,
+        detailVertexId: detailVertexId || undefined,
+      }),
+    });
+    const myData = response;
+
+    myData.versions = myData.versions.map((version: any) => {
+      const relativeFirstSeen = formatTimestampToDuration(
+        version.firstSeen,
+        CALCULATE_DURATION_SINCE
+      );
+      const relativeLastSeen = formatTimestampToDuration(
+        version.lastSeen,
+        CALCULATE_DURATION_SINCE
+      );
+
+      const fudgeFactorSeconds = 2 * minIntervalSeconds;
+      const isLastSeenCloseToNow = Date.now() - version.lastSeen <= fudgeFactorSeconds * 1000;
+
+      return {
+        ...version,
+        relativeFirstSeen: i18n.translate(
+          'xpack.monitoring.logstash.pipeline.relativeFirstSeenAgoLabel',
+          {
+            defaultMessage: '{relativeFirstSeen} ago',
+            values: { relativeFirstSeen },
+          }
+        ),
+        relativeLastSeen: isLastSeenCloseToNow
+          ? i18n.translate('xpack.monitoring.logstash.pipeline.relativeLastSeenNowLabel', {
+              defaultMessage: 'now',
+            })
+          : i18n.translate('xpack.monitoring.logstash.pipeline.relativeLastSeenAgoLabel', {
+              defaultMessage: 'until {relativeLastSeen} ago',
+              values: { relativeLastSeen },
+            }),
+      };
+    });
+    setData(myData);
+    updateTotalItemCount(response.totalNodeCount);
+  }, [ccs, clusterUuid, services.data?.query.timefilter.timefilter, services.http, getPaginationRouteOptions, updateTotalItemCount]);
+
+  useEffect(() => {
+    if (data.pipeline) {
+      console.log('data.pipeline', data.pipeline)
+      setPipelineState(new PipelineState(data.pipeline))
+    }
+  }, [data])
+
+  const timeseriesTooltipXValueFormatter = (xValue: any) => moment(xValue).format(dateFormat);
+
+  const onVertexChange = useCallback((vertex: any) => {
+    if (!vertex) {
+      setDetailVertexId(null)
+    } else {
+      setDetailVertexId(vertex.id)
+    }
+
+    getPageData()
+  }, [getPageData]);
+
+
+  return (
+    <LogstashTemplate
+      title={title}
+      pageTitle={pageTitle}
+      getPageData={getPageData}
+      cluster={cluster}
+    >
+      <div>
+        {pipelineState && <PipelineViewer
+          pipeline={List.fromPipeline(
+            Pipeline.fromPipelineGraph(pipelineState.config.graph)
+          )}
+          timeseriesTooltipXValueFormatter={timeseriesTooltipXValueFormatter}
+          setDetailVertexId={onVertexChange}
+          detailVertex={data.vertex ? vertexFactory(null, data.vertex) : null}
```

Comment:
Seems like something isn't right with this part that's causing the draw not to expand. Seems like detailVertex is always null.

I can get it to display in the angular version

![Cursor_and_Screen_Shot_2021-10-04_at_10_54_12](https://user-images.githubusercontent.com/690/135783539-86e8037b-66af-4aa4-97e3-38f9bd9e603f.png)